packages/backend-ai/src/conversation/factory.test.ts

```diff
@@ -0,0 +1,125 @@
+import { beforeEach, describe, it, mock } from 'node:test';
+import { App, Stack } from 'aws-cdk-lib';
+import {
+  ConstructFactoryGetInstanceProps,
+  ResourceNameValidator,
+} from '@aws-amplify/plugin-types';
+import assert from 'node:assert';
+import { StackMetadataBackendOutputStorageStrategy } from '@aws-amplify/backend-output-storage';
+import {
+  ConstructContainerStub,
+  ResourceNameValidatorStub,
+  StackResolverStub,
+} from '@aws-amplify/backend-platform-test-stubs';
+import { defaultEntryHandler } from './test-assets/with-default-entry/resource.js';
+import { customEntryHandler } from './test-assets/with-custom-entry/resource.js';
+import { Template } from 'aws-cdk-lib/assertions';
+import { defineConversationHandlerFunction } from './factory.js';
+
+const createStackAndSetContext = (): Stack => {
+  const app = new App();
+  app.node.setContext('amplify-backend-name', 'testEnvName');
+  app.node.setContext('amplify-backend-namespace', 'testBackendId');
+  app.node.setContext('amplify-backend-type', 'branch');
+  const stack = new Stack(app);
+  return stack;
+};
+
+void describe('ConversationHandlerFactory', () => {
+  let rootStack: Stack;
+  let getInstanceProps: ConstructFactoryGetInstanceProps;
+  let resourceNameValidator: ResourceNameValidator;
+
+  beforeEach(() => {
+    rootStack = createStackAndSetContext();
+
+    const constructContainer = new ConstructContainerStub(
+      new StackResolverStub(rootStack)
+    );
+
+    const outputStorageStrategy = new StackMetadataBackendOutputStorageStrategy(
+      rootStack
+    );
+
+    resourceNameValidator = new ResourceNameValidatorStub();
+
+    getInstanceProps = {
+      constructContainer,
+      outputStorageStrategy,
+      resourceNameValidator,
+    };
+  });
+
+  void it('creates singleton instance', () => {
+    const factory = defaultEntryHandler;
+    const instance1 = factory.getInstance(getInstanceProps);
+    const instance2 = factory.getInstance(getInstanceProps);
+    assert.strictEqual(instance1, instance2);
+  });
+
+  void it('resolves default entry when not specified', () => {
+    const factory = defaultEntryHandler;
+    const lambda = factory.getInstance(getInstanceProps);
+    const template = Template.fromStack(Stack.of(lambda.resources.lambda));
+    template.resourceCountIs('AWS::Lambda::Function', 1);
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      Handler: 'index.handler',
+    });
+  });
+
+  void it('uses entry that is explicitly specified', () => {
+    const factory = customEntryHandler;
+    const lambda = factory.getInstance(getInstanceProps);
+    const template = Template.fromStack(Stack.of(lambda.resources.lambda));
+    template.resourceCountIs('AWS::Lambda::Function', 1);
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      Handler: 'index.handler',
```

Comment:
shouldn't this be something like index.custom_handler from https://github.com/aws-amplify/amplify-backend/pull/1858/files#diff-bc30defca6e6547d73100e297149be6f8eb1940b1ec0a5ecaf67b311e8511624R9

Reply:
`Handler: 'index.handler',` is not related to file name.

`index` is there because esbuild bundles everything into single `index.js` . This happens no matter how input sources are defined.

`handler` is expected export from `index.js`. I.e. relates to `export const handler = async () => {` .

Reply:
gotcha, then how is this test validating that the correct entry is used? is there any other attribute that can be matched

Reply:
We can't do that at this level of testing, so this test is kind of best effort. It will surface some bugs when entry resolution points to non existing files.
Otherwise this aspect is also covered in e2e tests where outcomes from lambdas are used in assertions.

See https://github.com/aws-amplify/amplify-backend/blob/4fb161cef9fae2cad5d2ae2e271eefd27470ba29/packages/backend-function/src/factory.test.ts#L141-L143